### DISCUSSION TITLE
Help to save the following structure than react-hook-form

### DISCUSSION THREAD

**Discussion by CRAG666** [react-hook-form/react-hook-form#7665] [Q&A] [answered]
I am creating a dynamic form (the creation is already solved) but the answers of all these questions must be stored in a structure like this:
``` typescript
[
  {
     id: 1,
     answer: "yes"
  },
  {
     id: 2,
     answer: "no"
  {
]
```
The question is how could I achieve this?
I've read about `useFieldArray` but the most I can do is just store a similar structure but without the ids, the ids are static and given by an api. to insert the answers it is done with the structure that I already mentioned above.
I would greatly appreciate your help

**Comment by bluebill1049** [comment, marked as answer]
have a look at this example: https://codesandbox.io/s/react-hook-form-usefieldarray-ssugn

**Comment by CRAG666** [comment]
It's exactly the example I saw, but that doesn't suit what I need. The structure that I show is the one that is needed generate. In the example that we mention there are 1 object with 2 fields  firstName and lastName which refer to two inputs, in my case it is a single input that stores two values, the id and its answer. For example, in one of my attempts, I changed this line to this:

```typescript
// in the example
test: [{ firstName: "Bill", lastName: "Luo" }]
// my code
test: [{ id: 1, answer: "" }, { id: 2, answer: "" }, ...]
```
but when capturing the responses it only returns this:
```typescript
[ 
  { 
     respuesta : "sí" 
  } , 
  { 
     respuesta : "no" 
  { 
]
```
Maybe the solution is easy but I haven't been able to solve this for days :(

**Reply by bluebill1049** [reply]
oh, you want to keep the `id`? I believe the latest version will keep your `id` field.

**Reply by Moshyfawn** [reply]
Yep, check your RHF version. Since [ver. 7.23.0](https://github.com/react-hook-form/react-hook-form/releases/tag/v7.23.0) RHF shouldn't override the ID field of your data, if you have one. You'll still see the generated by the lib ID in place of your data ID when iterating over the fields array, but the submitted data will have exactly your data structure with your data iD.

**Reply by CRAG666** [reply]
thanks

**Comment by CRAG666** [comment]
thanks its worked, but i have new problem, please help me.
https://github.com/react-hook-form/react-hook-form/discussions/7689
